### ISSUE TITLE
Output Capability issues

### ISSUE THREAD

**Issue by davqvist** [Frinn38/Custom-Machinery#6]
I am having a lot of issues with capabilities, I will group them in one issue, let me know if I should split them up. Also let me know if you need jsons to test.

- Energy: Created energy doesn't get output. You have to pull the energy to get it out what most cables don't do. I could only get it done with Mekanism cables (configurated).
- Fluids: I can't get fluids to get extracted from machines at all, even if you pull, e.g. Mekanism pipes.
- Items: placing a hopper under a machine voids items if they are created afterwards. Only if they were already present in the machine they get extracted correctly.

Forge 36.1.18
CM 0.1.3

**Comment by Frinn38** [comment]
Ok so first this is completely normal that the machine doesn't auto insert or extract any capablity (including energy) because this functionality is not present yet, I was planning to implement that at the same time I work on the facing i/o modes.

For energy I guess I can make it to auto extract from all sides for now, so it will put energy on connected cables, the drawback would be that for machine that only consume FE, if you didn't set maxOutput to 0 it will still try to extract energy on the cable that insert energy into the machine... maybe I will just add the "mode" property to the energy component so you can set it input or output only (or both or none).

For the fluid capability, I found some broken logic on the fluid component fill/drain from external tiles, which caused the extracted fluid amount to be incorrect. However I could still extract and insert fluid with mek pipes even if the amount was incorrect. The incorrect amount bug should be fixed, when the next update is pushed let me know if this also fixed your bug.

For items, I cannot reproduce this issue, I tried with the Power Crusher included in the test datapack and the items were correctly extracted by the hopper. Send me your machine json here or on discord so I can test with exactly the same machine.

**Comment by davqvist** [comment]
- I should have clarified, sorry. I know that machines don't auto extract, but for energy, this should probably be the default as most mods don't extract energy with their connectors or cobles. What you are suggesting, is exactly what should be default: energy should autoextract on all sides if maxOutput>0

- I definitely can't extract liquids from this simple machine. https://pastebin.com/mhF4NEb6 It has no gui because it is not needed, but The One Probe shows me that it works. this is the recipe: https://pastebin.com/2s1Dt5Ey

- For items, try the Stone Gen from the test pack, that one doesn't work.

**Comment by Frinn38** [comment]
Ok so I confirm there is an issue with fluids and items capabilities. 
Problem is the issue only show outside of my dev environment so this will be a lot harder to debug.

EDIT:
It look like the fluid issue is gone with 0.1.4, at least I can't reproduce it even outside my dev env. let me know if that's not the case for you.
I found something interesting about the item issue, the items are void only if 1 is created. If the recipe create 2 items in the same stack, both items are perfectly extracted by the hopper... Will do more testing tomorrow but it may be just some brainfart of me in the item component logic.

EDIT 2:
Found the item issue, should be fixed in 0.1.4
